### ISSUE TITLE
First suggestions

### ISSUE THREAD

**Issue by GWPvGWP** [Vexcited/StayReal#6]
Hi,

The app looks promising -and so tidy-, thank you!
I know you are still developing, so no pressure intended. May I just propose a few things I see not yet on the feature list? Regard it as a sign of appreciation ;)

-Hiding the second image, so you can see the full image. For example by a single tap (preferred), or by holding the image like in the original app.
-Making the images zoomable
-Support for videos
-Option to access the full image (.webp) by getting the url or download. Same for videos if that's getting supported.
-Giving the amount of friends that have posted
-Discern between 'last moment' today or yesterday. "Last moment was at 10:00" could mean both when you look at 11:00.

-When local caching succeeds it would be fun to have to ability to save some posts
-When friends listing succeeds it would be nice to have sorting (on name, date added). Of minor importance, but if it is easy to do, why not.

**Comment by Vexcited** [comment]
Hello, first of all thanks you for all these suggestions !

Most of the things you said are actually planned (but in my head), I forgot to write them in the to-do lists, so I guess I'll pin your issue to make sure I don't forget them.

I'm actually in a big rewrite phase (which includes a native tauri plugin to call kotlin code) for supporting notifications and background updates, which is kinda important for caching !

---

> When local caching succeeds it would be fun to have to ability to save some posts

I plan to save all posts by default : the whole `feeds` cache will be stored day by day.
No image will be saved, only the URLs (since they're persistent, there's no reason to save the images - except for offline usage : you'd still be able to save the images locally but that'll be a user action and no default)

**Comment by Vexcited** [comment]
Also some pages are still in development, like the `/friends`, `/settings` (which gives a black screen at the moment) and the `/profile` page where all the information is shown as text without any styling... I was kinda lazy to do it to be honest :')

**Comment by GWPvGWP** [comment]
Perfect, that's what I wanted to hear :)

Op vr 15 nov 2024 om 16:26 schreef Mikkel ALMONTE--RINGAUD <
***@***.***>:

**Comment by Vexcited** [comment]
What do you mean `via localhost` ?

**Comment by Vexcited** [comment]
No it's not possible because BeReal's API does not allow CORS. You'd need a web proxy in between.

Anyway you'd still have to rewrite a huge part of the app because in my current development branch (the notifications one) I'm rewriting a big part of my internal API in Kotlin and Rust for a more in-deep integration with Tauri.

**Comment by Vexcited** [comment]
Thanks you ! It'll be done as soon as I get notifications to work, I'm currently spending all my time on this (https://github.com/Vexcited/StayReal/pull/7)

**Comment by GWPvGWP** [comment]
Hiding the second image by tap-and-hold did work in previous versions, but currently not anymore. Don't know if it's only my device.

**Comment by Vexcited** [comment]
Yes, I commented it only the issue linked to it : https://github.com/Vexcited/StayReal/issues/12#issuecomment-2565778240

**Comment by Vexcited** [comment]
By the way...

> Option to access the full image (.webp) by getting the url or download

is now implemented, you can open the images in your system's browser by clicking the three dots next to a post in the latest versions.

**Comment by Vexcited** [comment]
Also, I'm going to close this issue since I splitted it into several other issues, would be better for tracking. If there's something you want I didn't make an issue for, feel free to open one.

But make sure to separate the features and not group everything in a single issue.
Thanks you !
